ConnectionManager.lib.nut

```diff
@@ -49,13 +69,19 @@ class ConnectionManager {
         // Set the timeout policy + disconnect if required
         server.setsendtimeoutpolicy(RETURN_ON_ERROR, WAIT_TIL_SENT, ackTimeout);
 
-        if (startDisconnected) {
-            // Disconnect if required
-            server.disconnect();
-            _connected = false;
-        } else if (startConnected) {
-            // Start connecting if they ask for it
-            imp.wakeup(0, connect.bindenv(this));
+        switch (startBehavior) {
+        	case START_NO_ACTION:
+	        	// Do nothing
+	        	break;
+        	case START_CONNECTED:
+	        	// Start connecting if they ask for it
+	            imp.wakeup(0, connect.bindenv(this));
+	            break;
+        	case START_DISCONNECTED:
+	        	// Disconnect if required
+	            server.disconnect();
```

Comment:
I believe a) it should be `disconnect()` as it takes care of flushing the cache and some other stuff before disconnect b) it should be called the same way as connect is called above.